kernel/frame_allocator/src/lib.rs

```diff
@@ -861,56 +1038,66 @@ fn contains_any(
+    // first check the regions list for overlaps and proceed only if there are none.
+    if !contains_any(&regions_list, &frames){
+        // Check whether the reserved region overlaps any existing regions.
+        match &mut frames_list.0 {
+            Inner::Array(ref mut arr) => {
+                for chunk in arr.iter().flatten() {
+                    if let Some(_overlap) = chunk.overlap(&frames) {
+                        // trace!("Failed to add reserved region {:?} due to overlap {:?} with existing chunk {:?}",
+                        //     frames, _overlap, chunk
+                        // );
+                        return Err("Failed to add free frames that overlapped with existing frames (array).");
+                    }
                 }
             }
-        }
-        Inner::RBTree(ref mut tree) => {
-            let mut cursor_mut = tree.upper_bound_mut(Bound::Included(frames.start()));
-            while let Some(chunk) = cursor_mut.get().map(|w| w.deref()) {
-                if chunk.start() > frames.end() {
-                    // We're iterating in ascending order over a sorted tree,
-                    // so we can stop looking for overlapping regions once we pass the end of the new frames to add.
-                    break;
-                }
-                if let Some(_overlap) = chunk.overlap(&frames) {
-                    // trace!("Failed to add reserved region {:?} due to overlap {:?} with existing chunk {:?}",
-                    //     frames, _overlap, chunk
-                    // );
-                    return Err("Failed to add reserved region that overlapped with existing reserved regions (RBTree).");
+            Inner::RBTree(ref mut tree) => {
+                let mut cursor_mut = tree.upper_bound_mut(Bound::Included(frames.start()));
+                while let Some(chunk) = cursor_mut.get().map(|w| w.deref()) {
+                    if chunk.start() > frames.end() {
+                        // We're iterating in ascending order over a sorted tree,
+                        // so we can stop looking for overlapping regions once we pass the end of the new frames to add.
+                        break;
+                    }
+                    if let Some(_overlap) = chunk.overlap(&frames) {
+                        // trace!("Failed to add reserved region {:?} due to overlap {:?} with existing chunk {:?}",
+                        //     frames, _overlap, chunk
+                        // );
+                        return Err("Failed to add free frames that overlapped with existing frames (RBTree).");
+                    }
+                    cursor_mut.move_next();
                 }
-                cursor_mut.move_next();
             }
         }
+        
+        regions_list.insert(PhysicalMemoryRegion {
+                typ: MemoryRegionType::Reserved,
+                frames: frames.clone(),
+        }).map_err(|_c| "BUG: Failed to insert non-overlapping frames into list.")?;
```

Comment:
minor semantic correction for accuracy:
```suggestion
        }).map_err(|_c| "BUG: Failed to insert non-overlapping physical memory region into reserved regions list.")?;
```